File: dotnet/animals/Cow.cs
using System;

namespace dotnet
{
    ///<sumary>
    /// Author: brayvasq <brayvasq@gmail.com>
    /// Version: 0.1
    /// Date: 2018-03-25
    /// This is a class that represent a Cow inheriting animal class.
    ///</sumary>
    public class Cow : Animal{
        ///<sumary>This methods is the constructor of the class and use the parent constructor.</sumary>
        ///<param name="id">is a identifier of the cow.</param>  
        ///<param name="name">is a name for the cow.</param>  
        ///<param name="age">is an age that is used to determinate if is alive</param>  
        ///<param name="weight"></param>          
        ///<param name="price">is the simple price of the cow</param>
        public Cow(string id,string name,int age, float weight,double price) : base(id,name,age,weight,price){
            this.alive = this.age <= 25 ? true:false;
        }
        ///<sumary>This is a overwritten method from the parent class.</sumary>
        /// <returns>final price based on a simple price and the weight of the animal</returns>
        public override double calculatePrice(){
            return this.price * 3 * this.weight;
        }
        ///<sumary>This is a overwritten method from the parent class.</sumary>
        ///<sumary>Just show the characteristic sound of the cow.</sumary>
        public override void speak(){
            Console.WriteLine("Muuuu!!");
        }
    }
}

File: golang/cmd/src/models/cow.go
// Package models contains the simple struct and interfaces
// for the representation of a general and specific animals of a farm
package models

import (
	"strconv"
	"strings"
	"time"

	"github.com/brayvasq/learn-languages/golang/cmd/src/utils"
)

// Cow contains behaviors and characteristics of a Cow
// It uses composition to use the FarmAnimal properties and behaviors
type Cow struct {
	FarmAnimal
}

// NewCow creates a new Cow instance,
// and setups the specific Cow properties
func NewCow(Name string) Cow {
	cow := Cow{}
	cow.Name = Name
	cow.Age = 0
	cow.LastWash = time.Now()
	cow.LastFood = time.Now()
	cow.Alive = true
	cow.LimitWash = 3
	cow.LimitFood = 3
	cow.Kind = "cow"
	return cow
}

// GetCharacter returns the Cow character.
// This method depends on the IsAlive() method
//
// This method was defined in the interface Animal.
// To implement the interface Animal, we just need to implement all the methods in the interface.
func (c *Cow) GetCharacter() string {
	character := "  (....)  \n"
	character += func() string { // This is an anonimus function
		if c.IsAlive() { // used to emulate a ternary operator
			return "（0 .. 0）"
		} else {
			return "（X .. X）"
		}
	}()
	return character
}

// Speak returns the Cow sound
//
// This method was defined in the interface Animal.
// To implement the interface Animal, we just need to implement all the methods in the interface.
func (c *Cow) Speak() string {
	return "Muuu Muuu!!!"
}

// GetInfo returns the Cow info.
// Show the cow character, all the specific properties
// And show how much time is left to give to the cow a shower or food without it having died
func (c *Cow) GetInfo() string {
	character := c.GetCharacter()
	animal := ""

	animal += strings.Repeat("#", 50) + "\n"
	animal += character + "\n"
	animal += "Name: " + c.Name + "\n"
	animal += "alive: " + strconv.FormatBool(c.Alive) + "\n"
	animal += "Sound: " + c.Speak() + "\n"

	// To show how much time have the Cow until it dies without showering,
	// just show the difference between the LimitWash and the result of
	// the  utils.Dates GetDiffInMinutes method for the LastWash, In that way we can know
	// how much the limit is exceeded
	animal += "Wash time limit: " + strconv.Itoa(int(c.LimitWash-utils.Dates{}.GetDiffInMinutes(c.LastWash))) + "\n"

	// To show how much time have the Cow until it dies without eat,
	// just show the difference between the LimitWash and the result of
	// the  utils.Dates GetDiffInMinutes method for the LastFood, In that way we can know
	// how much the limit is exceeded
	animal += "Food time limit: " + strconv.Itoa(int(c.LimitFood-utils.Dates{}.GetDiffInMinutes(c.LastFood))) + "\n"

	animal += strings.Repeat("#", 50) + "\n"

	return animal
}
